apps/site/lib/site_web/templates/schedule/_timetable.html.eex

```diff
@@ -155,6 +155,55 @@
             <% end %>
           <% end %>
         <% end %>
+        <%= if @route.id == "Orange" and @direction_id == 0 do %>
+         <% cell_background = if rem(2, 2) == 0 do "white" else "gray" end %>
+          <%= content_tag :tr, [class: stop_row_class(2)] do %>
+            <%= content_tag :th, [
+              scope: "row",
+              data: [absolute: true],
+              class: "js-tt-stop-name m-timetable__cell--first-column m-timetable__cell m-timetable__cell--" <>
+                cell_background
+            ] do %>
+              <div class="m-timetable__row-header m-timetable__stop-name notranslate">
+                Trains continues to
+              </div>
+            <% end %>
+            <td class="js-tt-cell hidden-no-js">
+              <div class="m-timetable__row-header"></div>
+            </td>
+            <%= for schedule <- @header_schedules do %>
+              <%
+                stop = Enum.at(@all_stops, 2)
+                trip_id = schedule.trip.id
+                trip_schedule = @trip_schedules[{trip_id, stop.id} || %{}]
+              %>
+              <td class="m-continue__row">
+                    <span class="m-timetable__schedule">
+                      <span class="m-timetable__time" >
+                      <%= if !is_nil(trip_schedule) do%>
+                        <%= V3Api.Trips.short_name(trip_schedule.trip.headsign) %>
+                      <% else %>
+                      <%# if trip schedule is nil check other stops for non nil value %>
+                        <%
+                          all_trips = for stop_idx <- [0,1,3] do
+                            stop = Enum.at(@all_stops, stop_idx)
+                            trip_id = schedule.trip.id
+                            trip_schedule = @trip_schedules[{trip_id, stop.id} || %{}]
```

Comment:
suggestion: These 3 lines seem essentially identical to lines 176–178 above, and in both cases, the only result we end up using is `trip_schedule`. In order to remove the repetition, I think that you could move this logic into a helper function in the `Timetable` view, and then be able to call it twice.

Reply:
Actually, as I look further, maybe even better than extracting just those 3 lines would be to extract all of the logic from line 183 through line 200 a function in the view. (Lines 175–179 would need to move as well, and you could still put those into a second, private helper function.) That all means that you would have a single function to call that would return the value of the `span` element.

The benefit I'd see in this refactor is that the page layout in the `.html.eex` file gets cleaner and easier to read, while at the same time you more explicitly define a chunk of business logic as a pure function with clear inputs and output as well as a name that helps describe the purpose. (This function is also easily testable in isolation if you were interested in that.)